rocketry/core/task.py

```diff
@@ -192,42 +193,37 @@ class Task(RedBase, BaseModel):
     ...         return ...
 
     """
-    class Config:
-        arbitrary_types_allowed= True
-        underscore_attrs_are_private = True
-        validate_assignment = True
-        json_encoders = {
-            Parameters: lambda v: v.to_json(),
-            'BaseCondition': lambda v: str(v),
-            FunctionType: lambda v: v.__name__,
-            'Session': lambda v: id(v),
-        }
-
+    model_config = ConfigDict(
+        arbitrary_types_allowed= True,
+        validate_assignment = True,
+        validate_default=True,
+        extra='allow',      
+    )
 
-    session: 'Session' = Field()
+    session: 'Session' = Field(validate_default=False, default=None)
 
     # Class
     permanent: bool = False # Whether the task is not meant to finish (Ie. RestAPI)
     _actions: ClassVar[Tuple] = ("run", "fail", "success", "inaction", "terminate", None, "crash")
     fmt_log_message: str = r"Task '{task}' status: '{action}'"
 
-    daemon: Optional[bool]
+    daemon: Optional[bool] = False
```

Comment:
If I recall correctly, if this is not set (should be ``None``), this is fetched from the session configs thus setting default here doesn't allow us to see that the user didn't want to specify the daemon attr for this particular task (thus should be gotten from the config).

Reply:
I see what you mean on this now
```
    def run_as_process(self, params:Parameters, direct_params:Parameters, task_run:TaskRun, daemon=None, log_queue: multiprocessing.Queue=None):
        """Create a new process and run the task on that."""

        session = self.session

        params = params.pre_materialize(task=self, session=session)
        direct_params = direct_params.pre_materialize(task=self, session=session)

        # Daemon resolution: task.daemon >> scheduler.tasks_as_daemon
        log_queue = session.scheduler._log_queue if log_queue is None else log_queue

        daemon = self.daemon if self.daemon is not None else session.config.tasks_as_daemon
```

Hopefully with a default value of None this should work now rather than False.